.github/workflows/ruff.yml

```
@@ -32,15 +29,10 @@ jobs:
       - name: Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
-      # Ruff does not need a specific python version installed
-      - name: Install dependencies
-        run: |
-          python -m pip install 'ruff==0.11.13'
-
       - name: Lint with Ruff
-        run: |
-          ruff check --target-version ${{ matrix.python-version }} --output-format=github
+        uses: astral-sh/ruff-action@v3
```

```suggestion
        uses: astral-sh/ruff-action@v3
        with:
          version-file: pyproject.toml
```
https://github.com/astral-sh/ruff-action?tab=readme-ov-file#install-a-version-from-a-specified-version-file